frontend/src/stateless/index.ts

```diff
@@ -0,0 +1,344 @@
+import { indexedDB as indexedDBtest } from 'fake-indexeddb';
+import * as jsyaml from 'js-yaml';
+import _ from 'lodash';
+import { v4 as uuidv4 } from 'uuid';
+import { request } from '../lib/k8s/apiProxy';
+import { Cluster } from '../lib/k8s/cluster';
+import { ConfigState, setStatelessConfig } from '../redux/configSlice';
+import store from '../redux/stores/store';
+
+/**
+ * Config is the object that is stored in localStorage as string format.
+ */
+interface Config {
+  [prop: string]: any;
+}
+
+/**
+ * KubeconfigObject is the object that is stored in indexDB as string format.
+ * It is a JSON encoded version of the kubeconfig file.
+ * It is used to store the kubeconfig for stateless clusters.
+ * This is basically a k8s client - go Kubeconfig object.
+ * @see storeStatelessClusterKubeconfig
+ * @see getStatelessClusterKubeConfigs
+ * @see findKubeconfigByClusterName
+ */
+interface KubeconfigObject {
+  apiVersion: string;
+  kind: string;
+  clusters: Array<{
+    name: string;
+    cluster: {
+      server: string;
+    };
+  }>;
+  users: Array<{
+    name: string;
+    user: {};
+  }>;
+  contexts: Array<{
+    name: string;
+    context: {
+      cluster: string;
+      user: string;
+    };
+  }>;
+  'current-context': string;
+}
+
+/**
+ * Store the kubeconfig for a stateless cluster in IndexedDB.
```

Comment:
Since we are storing **kubeconfig** in base64 encoded format, the localstorage has [limit](https://stackoverflow.com/questions/2989284/what-is-the-max-size-of-localstorage-values/) of storing ~5Mb of characters. And a user can create multiple stateless clusters which would only result in storage issues later on, so it was suggested and decided to use [indexDB](https://stackoverflow.com/questions/5692820/maximum-item-size-in-indexeddb) since there's isn't a specific limit for a size of a single item, only a global limit. And Maximum storage size depends on 2 things, Browser and Disk space.